forecasting/src/autogluon/forecasting/trainer/abstract_trainer.py

```diff
@@ -0,0 +1,668 @@
+        else:
+            model_names_trained = self._train_and_save(
+                train_data, model=model, val_data=val_data, time_limit=time_limit
+            )
+
+        return model_names_trained
+
+    def _train_and_save(
+        self,
+        train_data: TimeSeriesDataFrame,
+        model: AbstractForecastingModel,
+        val_data: Optional[TimeSeriesDataFrame] = None,
+        time_limit: Optional[float] = None,
+    ) -> List[str]:
+        """Fit and save the given model on given training and validation data and save the
+        trained model.
+
+        Returns
+        -------
+        model_names_trained: the list of model names that were successfully trained
+        """
+        fit_start_time = time.time()
+        model_names_trained = []
+        try:
+            if time_limit is not None:
+                if time_limit <= 0:
+                    logging.log(
+                        15, f"Skipping {model.name} due to lack of time remaining."
+                    )
+                    return model_names_trained
+            else:
+                logging.log(20, f"Fitting model: {model.name} ...")
+            model = self._train_single(
+                train_data, model, val_data=val_data, time_limit=time_limit
+            )
+            fit_end_time = time.time()
+
+            val_score = (
+                model.score(val_data) * METRIC_COEFFICIENTS[model.eval_metric]
+                if val_data is not None
+                else None
+            )
+            model.val_score = val_score
+
+            pred_end_time = time.time()
+
+            model.fit_time = model.fit_time or (fit_end_time - fit_start_time)
+            if model.predict_time is None:
+                model.predict_time = (
+                    None if val_score is None else (pred_end_time - fit_end_time)
+                )
+
+            self.save_model(model=model)
+
+        except Exception as err:
+            logger.error(
+                f"\tWarning: Exception caused {model.name} to fail during training... Skipping this model."
+            )
+            logger.error(f"\t\t{err}")
+            logger.exception("Detailed Traceback:")
+            del model
+        else:
+            self._add_model(model=model)  # noqa: F821
+            model_names_trained.append(model.name)  # noqa: F821
+            if self.low_memory:
+                del model
+
+        return model_names_trained
+
+    def _train_multi(
+        self,
+        train_data: TimeSeriesDataFrame,
+        hyperparameters: Optional[Union[str, Dict]] = None,
+        models: Optional[List[AbstractForecastingModel]] = None,
+        val_data: Optional[TimeSeriesDataFrame] = None,
+        hyperparameter_tune: bool = False,
+        time_limit: Optional[float] = None,
+    ) -> List[str]:
+        time_start = time.time()
+        if hyperparameters is not None:
+            self.hyperparameters = hyperparameters
```

Comment:
Careful: Imagine users being able to call fit multiple times (train 5 models in 1 fit call, then 7 more in another, etc.). Having a `self.hyperparameters` shouldn't be necessary.